### PR TITLE
Modernize the Bazel build tool-chain.

### PR DESCRIPTION
**Pull request by Kashomon**
This modernizes the build-tool chain:
- Switch to using go-mod
- Use Go 1.14.1
- Use new version of Bazel rules.

Fixes: https://github.com/GoogleCloudPlatform/k8s-cluster-bundle/issues/101

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -353,7 +353,7 @@ To make using Bazel easier, we use Gazelle to automatically write Build targets.
 To automatically write the Build targets, run:
 
 ```shell
-bazel run //:gazelle
+bazel run //:gazelle -- update-repos -from_file=go.mod
```
Comment (mikedanese) [inline README.md:356]: Suggest putting generated code in a separate file:
```
bazel run //:gazelle -- update-repos -from_file=go.mod -to_macro 'repositories.bzl%gomod_repositories'
```

File: go.sum [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+k8s.io/apiextensions-apiserver v0.0.0-20181204003618-e419c5771cdc h1:IOukeE9HtTwpLslbujLDfRpfFU6tsjq28yO0fjnl/hk=
+k8s.io/apiextensions-apiserver v0.0.0-20181204003618-e419c5771cdc/go.mod h1:IxkesAMoaCRoLrPJdZNZUQp9NfZnzqaVzLhb2VEQzXE=
+k8s.io/apimachinery v0.0.0-20190208202428-1a579f8a7b42 h1:ju2lLx7i6XE8A9QLtwxlQngelP8SosMIjK4IYE/TLFI=
+k8s.io/apimachinery v0.0.0-20190208202428-1a579f8a7b42/go.mod h1:ccL7Eh7zubPUSh9A3USN90/OzHNSVN6zxzde07TDCL0=
+k8s.io/client-go v0.0.0-20190117233410-4022682532b3 h1:7VVBo3+/iX6dzB8dshNuo6Duds/6AoNP5R59IUnwoxg=
+k8s.io/client-go v0.0.0-20190117233410-4022682532b3/go.mod h1:7vJpHMYJwNQCWgzmNV+VYUl1zCObLyodBc8nIyt8L5s=
+k8s.io/klog v0.2.1-0.20190222023857-8145543d67ad h1:JbtnY2vNEOq+ovwkInIrO6z2AM84vtMJjwxPFcZLg3w=
+k8s.io/klog v0.2.1-0.20190222023857-8145543d67ad/go.mod h1:Gq+BEi5rUBO/HRz0bTSXDUcqjScdoY3a9IHpCEIOOfk=
+k8s.io/kube-openapi v0.0.0-20181114233023-0317810137be h1:aWEq4nbj7HRJ0mtKYjNSk/7X28Tl6TI6FeG8gKF+r7Q=
+k8s.io/kube-openapi v0.0.0-20181114233023-0317810137be/go.mod h1:BXM9ceUBTj2QnfH2MK1odQs778ajze1RxcmP6S8RVVc=
```
Comment (mikedanese) [inline go.sum:153]: What versions are these? Want to pin to an acutal release?

Comment (Kashomon) [reply go.sum:153]: shrug. I just did whatever go mod did. how do I pin to an actual release? =)

File: README.md (hunk 1 above)
Comment (Kashomon) [reply README.md]: Fixed

File: go.sum (hunk 2 above)
Comment (Kashomon) [reply go.sum:153]: Or, should I, even?

Comment (mikedanese) [reply go.sum:153]: You probably want to eventually. I suppose we can do that separately though.

**Comment by mikedanese** [comment]
LGTM

**Pull request merged**